cloudify_agent/api/pm/base.py

```diff
@@ -274,6 +287,22 @@ def __init__(self, logger=None, **params):
                               backend=self.broker_url)
         self._celery.conf.update(
             CELERY_TASK_RESULT_EXPIRES=defaults.CELERY_TASK_RESULT_EXPIRES)
+        if self._get_ssl_cert_path() != '':
+            self._celery.conf.BROKER_USE_SSL = {
+                'ca_certs': self._get_ssl_cert_path(),
+                'cert_reqs': ssl.CERT_REQUIRED,
+            }
+
+    def _create_celery_conf(self):
+        utils.render_template_to_file(
+            template_path='pm/shared/worker_conf.py.template',
+            file_path=os.path.join(self.virtualenv,
```

Comment:
Agreed- at best it's a bit ugly, and at worst it could fail interestingly. The problem with both this and the dynamic creation of the cert file are that I can't currently see a safer, saner, and (importantly) working alternative.